scripts/core/superdesk-api.d.ts

```diff
@@ -2084,6 +2085,7 @@ declare module 'superdesk-api' {
     export interface IPropsWidgetHeading {
         widgetName: string;
         editMode: boolean;
+        widgetId: string;
```

Comment:
Why do you need widget id? How about using label is unique identifier? It's shouldn't be supported to have multiple widgets using the same name anyway.

Reply:
I need it, in order to calculate properly the `pinned` state in another component. Using labels as unique IDs doesn't sound right to me, that's why I added `widgetId`

Reply:
> doesn't sound right

Doesn't sound like an argument to me :smile:

I won't ask you to undo this, but let's not do this in the future. If we have a value that is already unique it doesn't make sense to add another. It might be said that labels might not be unique - but they should in this and all similar cases. To enforce it, I would rather add a check on appStart, throw an error and refuse to start the app if some values that are supposed to be unique - are not. If we don't do this, the app technically works, but users see 2 options to click that have the same name, but do different things - which also must not happen.

Reply:
Ok, I can remove the ID and use name. Also if I do that I'll need to do extra modifications so that `getPinnedWidget` in `authoring-react.tsx` can return the widget name. Currently it only can access `id` since that's what's stored in state.